ambry-router/src/test/java/com.github.ambry.router/NonBlockingRouterTest.java

```diff
@@ -395,6 +395,50 @@ public void onBlobDeleted(String blobId) {
   }
 
   /**
+   * Test that if a composite blob is deleted, the data chunks are eventually deleted.
+   */
+  @Test
+  public void testCompositeBlobDataChunksDelete() throws Exception {
+    // Ensure there are 4 chunks.
+    maxPutChunkSize = PUT_CONTENT_SIZE / 4;
+    Properties props = getNonBlockingRouterProperties("DC1");
+    VerifiableProperties verifiableProperties = new VerifiableProperties((props));
+    MockClusterMap mockClusterMap = new MockClusterMap();
+    MockTime mockTime = new MockTime();
+    MockServerLayout mockServerLayout = new MockServerLayout(mockClusterMap);
+    // metadata blob + data chunks.
+    final CountDownLatch deletesDoneLatch = new CountDownLatch(5);
+    LoggingNotificationSystem deleteTrackingNotificationSystem = new LoggingNotificationSystem() {
+      @Override
+      public void onBlobDeleted(String blobId) {
+        deletesDoneLatch.countDown();
+      }
+    };
+    router = new NonBlockingRouter(new RouterConfig(verifiableProperties), new NonBlockingRouterMetrics(mockClusterMap),
+        new MockNetworkClientFactory(verifiableProperties, mockSelectorState, MAX_PORTS_PLAIN_TEXT, MAX_PORTS_SSL,
+            CHECKOUT_TIMEOUT_MS, mockServerLayout, mockTime), deleteTrackingNotificationSystem, mockClusterMap,
+        mockTime);
+
+    setOperationParams();
+
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
+    router.deleteBlob(blobId, null).get();
+
+    long waitStart = SystemTime.getInstance().milliseconds();
+    while (router.getBackgroundOperationsCount() != 0
+        && SystemTime.getInstance().milliseconds() < waitStart + AWAIT_TIMEOUT_MS) {
+      Thread.sleep(1000);
```

Comment:
not sure why you need this when you are doing `deletedDoneLatch.await()`. Could you explain?